### ISSUE TITLE
TLSContext cannot find secrets without Legacy Mode from another namespace

### ISSUE THREAD

**Issue by lawliet89** [emissary-ingress/emissary#3404]
**Describe the bug**

Without legacy mode enabled on 1.13, Ambassador complains that it cannot find the Kubernetes secrets containing the TLS certificates defined in `TLSContext` that refers to secrets in a different namespace.

For example:

```yaml
apiVersion: getambassador.io/v2
kind: TLSContext
metadata:
  annotations:
    meta.helm.sh/release-name: app-ui-staging
    meta.helm.sh/release-namespace: app-staging
  labels:
    app.kubernetes.io/instance: app-ui-staging
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: app-ui
    helm.sh/chart: app-ui-0.6.0
  name: app-ui-staging-0
  namespace: app-staging
spec:
  alpn_protocols: h2, http/1.1
  ambassador_id:
  - external
  hosts:
  - example.ai
  min_tls_version: v1.2
  secret: example-ai-wildcard.core
```

```
2021-04-21 03:10:36 diagd 1.13.0 [P20TAEW] INFO: starting Scout checker and timer logger
2021-04-21 03:10:36 diagd 1.13.0 [P20TAEW] INFO: starting event watcher
2021-04-21 03:10:40 diagd 1.13.0 [P20TAEW] ERROR: Secret example-ai-wildcard.core unknown
2021-04-21 03:10:40 diagd 1.13.0 [P20TAEW] INFO: app-api-staging-0.app-staging.1: <RichStatus BAD error='TLSContext app-api-staging-0 found no certificate in secret example-ai-wildcard in namespace core, ignoring...' hostname='ambassador-external-6c4f79dbc8-8qrnz' version='1.13.0'>
2021-04-21 03:10:40 diagd 1.13.0 [P20TAEW] ERROR: Secret example-ai-wildcard.core unknown
2021-04-21 03:10:40 diagd 1.13.0 [P20TAEW] INFO: app-ui-staging-0.app-staging.1: <RichStatus BAD error='TLSContext app-ui-staging-0 found no certificate in secret example-ai-wildcard in namespace core, ignoring...' hostname='ambassador-external-6c4f79dbc8-8qrnz' version='1.13.0'>
2021-04-21 03:10:40 diagd 1.13.0 [P20TAEW] ERROR: Secret example-ai-wildcard.core unknown
2021-04-21 03:10:40 diagd 1.13.0 [P20TAEW] INFO: app-ui-staging-redir-example.app-staging.1: <RichStatus BAD error='TLSContext app-ui-staging-redir-example found no certificate in secret example-ai-wildcard in namespace core, ignoring...' hostname='ambassador-external-6c4f79dbc8-8qrnz' version='1.13.0'>
```

If legacy mode is enabled, everything works fine.

When I `exec` into the container, I can see that Ambassador only finds the TLS secrets that are used in TLS Origination, but not for TLS termination:

```
/ambassador/snapshots/core/secrets-decoded $ ls -al
total 12
drwxr-xr-x    3 ambassad root          4096 Apr 23 01:38 .
drwxr-xr-x    3 ambassad root          4096 Apr 23 01:38 ..
drwxr-xr-x    2 ambassad root          4096 Apr 23 01:38 ambassador-consul-connector-internal
```

The secrets are also missing in the snapshot file.

I can confirm that the RBAC for the pod is correct. If I use `kubectl` from within the container, it can get the secrets.

**Versions (please complete the following information):**
 - Ambassador: [e.g. 0.32.1] 1.13.x
 - Kubernetes environment [e.g. Minikube, bare metal, Google Kubernetes Engine] GKE
 - Version [e.g. 1.8.1] 1.18

**Comment by lawliet89** [comment]
@esmet Please let me know if I've missed any details.

**Comment by lawliet89** [comment]
I've fixed this by upgrading the CRD.

I installed Ambassador via the Helm Chart and Helm [does not manage the lifecycle of CRDs](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/). 

Relevant issue: https://github.com/datawire/ambassador/issues/3417

**Comment by kflynn** [comment]
Sorry for the delay here! I note that `secret_namespacing` isn't set in your `TLSContext` -- do you by chance have `tls_secret_namespacing` set in your `ambassador` `Module`? One of those must be set as of... geeez... 1.7 or something? for cross-namespace secrets to work this way.

**Comment by lawliet89** [comment]
@kflynn No. I don't think these were in the documentation so I did not enable them.

In any case, I managed to fix this with the following findings:

- I used a different Consul Connector TLS secret for TLS origination that was created by a different `ambassadaor_id`. Using the "right one" fixed TLS orgination for Consul Connect
- I found out that you cannot add `.namespace` when the secret is in the same namespace as the `TLSContext`. Removing it fixed the issue.

Everything else works.
